cosmos/dbt/project.py

```diff
@@ -34,12 +36,36 @@ def has_non_empty_dependencies_file(project_path: Path) -> bool:
     return False
 
 
+def copy_dbt_packages(source_folder: Path, target_folder: Path) -> None:
+    """
+    Copies the dbt packages related files and directories from source_folder to target_folder.
+
+    :param: source_folder: The base directory where paths are sourced from.
+    :param: target_folder: The directory where paths will be copied to.
+    """
+    logger.info("Copying dbt packages to temporary folder...")
+    dbt_packages_paths = [DBT_PACKAGES_FOLDER, PACKAGE_LOCKFILE_YML]
+
+    for relative_path in dbt_packages_paths:
+        src_path = source_folder / relative_path
+        dst_path = target_folder / relative_path
+
+        os.makedirs(os.path.dirname(dst_path), exist_ok=True)
+
+        if src_path.is_dir():
+            shutil.copytree(src_path, dst_path, dirs_exist_ok=True)
+        else:
+            shutil.copy2(src_path, dst_path)
```

Comment:
Trying to understand what is the advantage of shutil.copy2 over shutil.copy here. Could you please share?

Reply:
`copy2` preserves the original file metadata - it probably does't make a difference here